app/services/recovery_code_generator.rb

```diff
@@ -1,7 +1,7 @@
 class RecoveryCodeGenerator
   attr_reader :user_access_key
 
-  def initialize(user, length: 16)
+  def initialize(user, length: 12)
```

Comment:
huh, yeah I like the idea of having this as an arg / default but setting the default value from a constant might be clearer...